### ISSUE TITLE
oas3-valid-media-example incorrect error when validating multipleOf 

### ISSUE THREAD

**Issue by JvNoordenburg** [stoplightio/spectral#2090]
**Describe the bug**
When using the `multipleOf` validation keyword, a valid number sometimes results in an error when using the spectral-cli

**To Reproduce**

1. Given this OAS3 document 
```yaml
openapi: 3.0.3
info:
  title: calculation-api
  description: |
    The calculation API.
  version: 0.0.1
  contact:
    name: test
    email: test@mail.com
tags:
  - name: Loan
    description: |
      Calculations related to loans.
paths:
  /loan-to-market-value-calculations:
    post:
      tags:
        - Loan
      summary: Calculates the loan to market value.
      description: |
        Calculates the loan to market value.
      operationId: CalculateLoanToMarketValue
      requestBody:
        $ref: '#/components/requestBodies/CalculateLoanToMarketValueRequest'
      responses:
        '200':
          $ref: '#/components/responses/CalculateLoanToMarketValueResponse'
  
components:  
  requestBodies:    
    CalculateLoanToMarketValueRequest:
      description: |
        The request body of CalculateLoanToMarketValue.
      required: true
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/LoanRealEstateLtmv'
          examples:
            calculate-loan-to-market-value:
              description: |
                The information of a loan and a real estate used to calculate the loan to market value.
              value:
                loan:
                  residualDebt: 4000.10
                realEstate:
                  marketValue: 3000.03  

  responses:   
    CalculateLoanToMarketValueResponse:
      description: |
        The successful response of CalculateLoanToMarketValue.
      content:
        application/json:
          schema:
            description: |
              The information of the calculated loan to market value.
            type: object
            required:
              - loanToMarketValue
            additionalProperties: false
            properties:
              loanToMarketValue:
                description: |
                  The calculated loan to market value.
                type: number
                format: decimal
                multipleOf: 0.000001
          examples:
            calculate-loan-to-market-value:
              description: |
                An example of the calculated loan to market value.
              value:
                loanToMarketValue: 75.001     
            another-calculate-loan-to-market-value:
              description: |
                Another example of the calculated loan to market value.
              value:
                loanToMarketValue: 75.0001  
            yet-another-calculate-loan-to-market-value:
              description: |
                Yet another example of the calculated loan to market value.
              value:
                loanToMarketValue: 75 

  schemas:        
    LoanRealEstateLtmv:
      description: |
        The information of a loan and a real estate used to calculate the loan to market value.
      type: object
      required:
        - loan
        - realEstate
      additionalProperties: false
      properties:
        loan:
          $ref: '#/components/schemas/LoanRealEstateLtmv.Loan'
        realEstate:
          $ref: '#/components/schemas/LoanRealEstateLtmv.RealEstate'

    LoanRealEstateLtmv.Loan:
      description: |
        The information of a loan used to calculate the loan to market value.
      type: object
      required:
        - residualDebt
      additionalProperties: false
      properties:
        residualDebt:
          description: |
            The total outstanding balance of the loan.
          type: number
          format: decimal
          multipleOf: 0.01

    LoanRealEstateLtmv.RealEstate:
      description: |
        The information of a real estate used to calculate the loan to market value.
      type: object
      required:
        - marketValue
      additionalProperties: false
      properties:
        marketValue:
          description: |
            The (estimated) market value of a real estate.
          type: number
          format: decimal
          multipleOf: 0.01
```

Ruleset:
```yaml
extends: 
- spectral:oas

# See https://meta.stoplight.io/docs/spectral/ZG9jOjExNw-open-api-rules
rules:

  #
  # OAS 3
  #

  # TODO: Make sure servers is always empty
  # OpenAPI servers must be present and non-empty array.
  oas3-api-servers: false

  # TODO: Ensure an example value is always present
  # Examples for requestBody or response examples can have an externalValue or a value, but they cannot have both
  oas3-examples-value-or-externalValue: true

  # Operation security values must match a scheme defined in the components.securitySchemes object
  oas3-operation-security-defined: true

  # TODO: Make sure servers is always empty
  # Server URL should not point at example.com.
  oas3-server-not-example.com: true

  # Server URL should not have a trailing slash.
  oas3-server-trailing-slash: true

  # Unused components should be removed
  oas3-unused-component: true

  # Examples must be valid against their defined schema.
  oas3-valid-media-example: true

  # Validate the structure of OAS v3
  oas3-schema: true

  # Parameters should have a description
  oas3-parameter-description: true
```

2. Run this CLI command 
```PowerShell
spectral lint ./test-contract.yaml --ruleset ./.spectral.yaml
```
4. See error:
```PowerShell
74:36  error  oas3-valid-media-example  "loanToMarketValue" property must be multiple of 0.000001  components.responses.CalculateLoanToMarketValueResponse.content.application/json.examples.calculate-loan-to-market-value.value.loanToMarketValue
```

**Expected behavior**
No errors. The value `75.001` is a valid value using the `multipleOf: 0.000001` keyword. 

**Environment (remove any that are not applicable):**
 - Library version: 6.3.0

**Additional context**
- There are some extra examples that do pass the validation so it isn't an issue for all validation rules
- I've also tested this with Spectral Studio resulting in the same errors (which makes sense as I assume this used the same command)

**Comment by P0lip** [comment]
I suppose that floating point precision plays a role here. I'm not really sure whether we can address it in Spectral.
This should probably be raised in the [ajv](https://github.com/ajv-validator/ajv) repo, because we use Ajv under the hood to perform that kind of validation.
I'll dig into it in a bit to see if that's indeed the case

**Comment by AlbertoFdzM** [comment]
> I suppose that floating point precision plays a role here. I'm not really sure whether we can address it in Spectral. This should probably be raised in the [ajv](https://github.com/ajv-validator/ajv) repo, because we use Ajv under the hood to perform that kind of validation. I'll dig into it in a bit to see if that's indeed the case

This is not likely an Ajv problem but a bad use of it. I've got this same error when using Ajv. You need to use the next config https://ajv.js.org/options.html#multipleofprecision

**Comment by JvNoordenburg** [comment]
Alright thanks. That's good to know. Is it possible to override the AJV options during creation of a ruleset?

**Comment by AlbertoFdzM** [comment]
> Alright thanks. That's good to know. Is it possible to override the AJV options during creation of a ruleset?

Nope, spectral do not allow to override the Ajv config it is using

**Comment by JvNoordenburg** [comment]
Ah well that's a shame but I think I can maybe work around it for now. Thanks!

**Comment by Amachua** [comment]
Hello, 

Have you scheduled to resolve this issue as we also encounter the issue with our configuration. 

To ease your investigation, I made a minimal reproduction case with:

- The specification `specification.yaml`:
```yaml
openapi: 3.0.0
info:
  title: Title.
  description: Description.
  version: 1.0.0
  contact:
    name: name
    email: email@email.com
    url: https://url.com
  license:
    name: Freemium
    url: https://url.com
paths: {}
servers:
  - url: https://url.com
tags:
  - name: TAG
    description: tag
components:
  schemas:
    validNumber:
      type: number
      multipleOf: 0.01
      example: 73.91
```

- The ruleset:
```yaml
extends:
  - spectral:oas
```

Running the command line `spectral lint specification.yaml` then I would like to only have the following feedback:
>  21:17  warning  oas3-unused-component      Potentially unused component has been detected.  components.schemas.validNumber

As we currently also have this one:
> 24:16    error  oas3-valid-schema-example  "example" property must be multiple of 0.01      components.schemas.validNumber.example

FWIW, it seems that this issue is also affecting the [prism repository](https://github.com/stoplightio/prism/issues).

**Comment by Mathouston** [comment]
I'm also encountering errors when using "multipleOf." Some values ​​are valid, while others are not.
Is there any way to prioritize this issue?
